### ISSUE TITLE
Dependency Prioritization 

### ISSUE THREAD

**Issue by chriscantu** [renovatebot/pro#57]
**What would you like Renovate to be able to do?**
**AS A** Pro (Enterprise) Renovate User
**I WANT TO** be able to prioritize some dependencies over others 
**SO THAT** I can ensure critical dependencies are updated first

**Describe the solution you'd like**
For large scale applications with lots of dependencies, we would like the ability to configure prioritization for certain dependencies which are more mission critical. In the case where a repo has set a limit for the # of PRs that can be submitted, renovate would close a lower priority PR when a more important is queued.  Also, Renovate should apply a label indicating the configured priority assigned to the dependency so it's clear for the maintainers which updates are more important. 

**Describe alternatives you've considered**
To date, we apply labels specifying our priority based on the "packagePatterns" .  However if our queue is full, new higher priority PRs are not submitted.  We want to be able to see those high priority updates as soon as possible. 

**Additional context**
N/A

**Comment by rarkins** [comment]
Which limit are you most concerned with? `prConcurrentLimit`?

**Comment by qswitcher** [comment]
I work with @chriscantu. Yes `prConcurrentLimit` is the limit we're referring to.

**Comment by rarkins** [comment]
How sure are you that you’d want to *close* existing PRs that no longer fit in the top X concurrent priority? I would think that could cause more noise than it saves at times. 

I really like the idea, trying to think how “Priority” can be applied genetically and widely for different use cases too.

**Comment by qswitcher** [comment]
What prioritization does Renovate do today when creating PRs? That is, if 100 dependencies are out of date, what criteria does it use to create PRs? especially if there is a `prConcurrentLimit`. What would be sufficient for us is if we could basically tell it "If you do a scan and there are 100 updates you need to make, but you only have room for 10 PRs, use these rules to sort based on priority".

Closing PRs would be a nice to have, but we could live without it just knowing that _when there is space available_, Renovate will only create PRs for the current highest priority package rules. Scheduling might work, but seems cumbersome since it would require playing games like: priority 1 gets scheduled every hour, priority 2 once a day, priority 3: once a week, etc, and would require a lot of tuning to make sure that priority 1 PRs _tend_ to fill up the `prConcurrentLimit` before lower priorities do.

**Comment by rarkins** [comment]
Currently prioritisation is done on update type: pin, digest, patch, minor, match. Ie raise the “easiest” ones first.

**Comment by chriscantu** [comment]
Another simple option instead of closing PR's is ignoring the limit and
always opening a top priority dependency.

**Comment by rarkins** [comment]
Yep, that's another good idea. BTW I think schedule and priority can live side by side too. Of course it would be complex if you really use both, but we should retain the ability to configure both.

**Comment by rarkins** [comment]
Moving this to the main repo as it will need to be implemented there as part of core
